### PR TITLE
Algebra in expression

### PR DESCRIPTION
**Pull request by denisshevchenko**
description
-----------

- [x] `+`, `-` and  `*` in monitor expression.
- [x] combinations of measurable and variable.
- [x] new tests.

checklist
---------

- [x] compiles (`cabal new-clean; cabal new-build`)
- [x] tests run successfully (`cabal new-test`)
- [x] documentation added and created (`cd docs; nix-shell --run make`)
- [x] link to an issue
- [x] link to an epic
- [x] add estimate points
- [x] add milestone (the same as the linked issue)

### CODE REVIEW

File: iohk-monitoring/test/Cardano/BM/Test/Monitoring.lhs [hunk 1]
```diff
@@ -41,14 +41,67 @@ unitTests = testGroup "Unit tests" [
...
+                      "parse and eval expression with algebra, variable + measurable; must return True" $
+                      parseEvalExpression "(  time<(stats.mean+(      10 s)      ))"
+                                          True
+                                          $ HM.fromList [ ("time",       Seconds 9)
+                                                        , ("stats.mean", Seconds 2)
+                                                        ]
+                , testCase
+                      "parse and eval expression with algebra, variable - variable; must return True" $
+                      parseEvalExpression "(time < (stats.mean-stats.min))"
+                                          True
```
Comment (CodiePP) [inline iohk-monitoring/test/Cardano/BM/Test/Monitoring.lhs]: can you duplicate the test cases and just also test for `False`?

File: iohk-monitoring/src/Cardano/BM/Output/Monitoring.lhs [hunk 2]
```diff
@@ -77,6 +82,7 @@ Function |effectuate| is called to pass in a |LogObject| for monitoring.
 instance IsEffectuator Monitor a where
     effectuate monitor item = do
         mon <- readMVar (getMon monitor)
+        effectuate (monBuffer mon) item
```
Comment (CodiePP) [inline iohk-monitoring/src/Cardano/BM/Output/Monitoring.lhs]: Now I got it! you have a local LogBuffer which records all values that are routed to monitoring.
Then, if the config does not contain a monitor expression for this item, you could avoid putting it into the queue. Or, maybe determine this early in the dispatcher as you do not have access to the configuration here.

File: iohk-monitoring/src/Cardano/BM/Output/Monitoring.lhs [hunk 3]
```diff
@@ -149,25 +168,88 @@ spawnDispatcher mqueue config sbtrace = do
...
+            _                                 -> []
+          where
+            stValues vn st =
+                [ Just (vn <> ".flast", flast st)
+                , Just (vn <> ".fold",  fold st)
+
+                , Just (vn <> ".fbasic.fmin",   fmin   . fbasic $ st)
+                , Just (vn <> ".fbasic.fmax",   fmax   . fbasic $ st)
+                , Just (vn <> ".fbasic.fcount", PureI . fromIntegral . fcount . fbasic $ st)
+                , Just (vn <> ".fbasic.fsum_A", PureD                . fsum_A . fbasic $ st)
```
Comment (CodiePP) [inline iohk-monitoring/src/Cardano/BM/Output/Monitoring.lhs]: access to the fields is great!
but, we do not need to provide access to `fsum_A` and `fsum_B` as they are just internally used to calculate "mean" and "stdev". can you add:
".fbasic.mean", PureD . meanOfStats . fbasic $ st
".fbasic.stdev", PureD . stdevOfStats . fbasic $ st
(or similar..)

File: iohk-monitoring/src/Cardano/BM/Output/Monitoring.lhs (hunk 2 above)
Comment (denisshevchenko) [reply iohk-monitoring/src/Cardano/BM/Output/Monitoring.lhs]: I did it by analogy with `Editor`: https://github.com/input-output-hk/iohk-monitoring-framework/blob/develop/iohk-monitoring/src/Cardano/BM/Output/Editor.lhs#L128

**Pull request merged**